brut.apktool/apktool-lib/src/main/java/brut/androlib/apk/YamlWriter.java

```diff
@@ -0,0 +1,122 @@
+package brut.androlib.apk;
+
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+
+public class YamlWriter implements Closeable {
+
+    private int mIndent = 0;
+    private final PrintWriter mWriter;
+    private final String QUOTE = "'";
+
+    public YamlWriter(OutputStream out) {
+        mWriter = new PrintWriter(new BufferedWriter(
+            new OutputStreamWriter(out, StandardCharsets.UTF_8)));
+    }
+
+    @Override
+    public void close() throws IOException {
+        mWriter.close();
+    }
+
+    public int getIndent() {
+        return mIndent;
+    }
+
+    public String getIndentString() {
+        // for java 11
+        // return " ".repeat(mIndent);
+        // for java 8
+        return String.join("", Collections.nCopies(mIndent, " "));
+    }
+
+    public static String escape(String value) {
+        return YamlStringEscapeUtils.escapeString(value);
+    }
+
+    public void nextIndent() {
+        mIndent += 2;
+    }
+
+    public void prevIndent() {
+        if (mIndent != 0)
+            mIndent -= 2;
+    }
+
+    public void writeIndent() {
+        mWriter.print(getIndentString());
+    }
+
+    public void writeInt(String key, int value) {
+        writeIndent();
+        mWriter.println(escape(key) + ": " + value);
+    }
+
+    public void writeBool(String key, boolean value) {
+        writeIndent();
+        String val = value ? "true": "false";
+        mWriter.println(escape(key) + ": " + val);
+    }
+
+    public void writeString(String key, String value, boolean quoted) {
+        writeIndent();
+        if (Objects.isNull(value)) {
+            mWriter.println(escape(key) + ": null");
+        } else {
+            if (quoted)
+                value = QUOTE + value + QUOTE;
+            mWriter.println(escape(key) + ": " + escape(value));
+        }
+    }
+
+    public void writeString(String key, String value) {
+        writeString(key, value, false);
+    }
+
+    public <T> void writeList(String key, List<T> list) {
+        if (Objects.isNull(list))
+            return;
+        writeIndent();
+        mWriter.println(escape(key) + ":");
+        for (T item: list) {
+            writeIndent();
+            mWriter.println("- " +  item);
+        }
+    }
+
+    public <K, V> void writeCommonMap(String key, Map<K, V> map) {
```

Comment:
This seems like a powerful function, but unused. Do we not have any maps at present to serialize/write?